tests/testthat/test-rec-with-table.R

```diff
@@ -0,0 +1,33 @@
+context("Integration Tests")
+
+test_that("Normal variables that depend on derived variables are properly recoded", {
+  data <- data.frame(
+    var_one_start = c(1, 2, 3)
+  )
+
+  variables_sheet <- read.csv(
+    file.path("../../assets/tests/integration/rec-with-table/variables-sheet.csv"),
+    fileEncoding = "UTF-8-BOM")
+
+  variable_details_sheet <- read.csv(
+    file.path(getwd(), "../../assets/tests/integration/rec-with-table/variable-details-sheet.csv"),
+    fileEncoding = "UTF-8-BOM")
+
+  variable_details_sheet$variableStart[3] <- "DerivedVar::[var_two]"
```

Comment:
@yulric It looks like this issue can be fixed by changing how we populate the details sheet.

The test was failing earlier because the `var_three` column was missing from the expected output. The problem was linked to the fact that the variable required to create `var_three` (i.e., `var_two`) was only present in the re-coded data, not the original.

You get the expected output when you change the value in `variableStart` from "var_two" to "DerivedVar::[var_two]".  A variable that relies on a derived variable is also a derived variable , regardless of whether it requires a function for re-coding. The data types of columns in the expected data were changed to factors to match the output, and `expect_equivalent()` was used to ignore attributes in the output data.

Reply:
Hmm, I don't think that should be the expected behavior. I think that the `DerivedVar` prefix should be used to specify that a variable is a derived variable i.e. a variable that requires a function for its recoding. 

For this issue, rather than changing the variable details sheet maybe we should change the way the code works so that "nested variables" also get recoded? So recodeflow would see that var_three depends on var_two which is not in the database, so it would pluck it out of the recoded data.